src/test/javascript/spec/component/exam/manage/exams/exam-detail.component.spec.ts

```diff
@@ -120,4 +173,141 @@ describe('ExamDetailComponent', () => {
         });
     }));
+
+    it('should correctly route to dashboard', fakeAsync(() => {
+        const location = examDetailComponentFixture.debugElement.injector.get(Location);
+        examDetailComponentFixture.detectChanges();
+        const dashboardButton = examDetailComponentFixture.debugElement.query(By.css('#tutor-exam-dashboard-button')).nativeElement;
+        dashboardButton.click();
+        examDetailComponentFixture.whenStable().then(() => {
+            expect(location.path()).to.equal('/course-management/1/exams/1/tutor-exam-dashboard');
+        });
+    }));
+
+    it('should correctly route to exercise groups', fakeAsync(() => {
+        const location = examDetailComponentFixture.debugElement.injector.get(Location);
+        examDetailComponentFixture.detectChanges();
+        const dashboardButton = examDetailComponentFixture.debugElement.query(By.css('#exercises-button-groups')).nativeElement;
+        dashboardButton.click();
+        examDetailComponentFixture.whenStable().then(() => {
+            expect(location.path()).to.equal('/course-management/1/exams/1/exercise-groups');
+        });
+    }));
+
+    it('should correctly route to scores', fakeAsync(() => {
+        const location = examDetailComponentFixture.debugElement.injector.get(Location);
+        examDetailComponentFixture.detectChanges();
+        const scoresButton = examDetailComponentFixture.debugElement.query(By.css('#scores-button')).nativeElement;
+        scoresButton.click();
+        examDetailComponentFixture.whenStable().then(() => {
+            expect(location.path()).to.equal('/course-management/1/exams/1/scores');
+        });
+    }));
+
+    it('should correctly route to students', fakeAsync(() => {
+        const location = examDetailComponentFixture.debugElement.injector.get(Location);
+        examDetailComponentFixture.detectChanges();
+        const studentsButton = examDetailComponentFixture.debugElement.query(By.css('#students-button')).nativeElement;
+        studentsButton.click();
+        examDetailComponentFixture.whenStable().then(() => {
+            expect(location.path()).to.equal('/course-management/1/exams/1/students');
+        });
+    }));
+
+    it('should correctly route to test runs', fakeAsync(() => {
+        const location = examDetailComponentFixture.debugElement.injector.get(Location);
+        examDetailComponentFixture.detectChanges();
+        const studentsButton = examDetailComponentFixture.debugElement.query(By.css('#testrun-button')).nativeElement;
+        studentsButton.click();
+        examDetailComponentFixture.whenStable().then(() => {
+            expect(location.path()).to.equal('/course-management/1/exams/1/test-runs');
+        });
+    }));
+
+    describe('test checkTotalPointsMandatory', () => {
+        it('should set totalPointsMandatory to false', () => {
+            examDetailComponent.exam = exam;
```

Comment:
These can be moved to beforeEach, since it gets repeated.

Reply:
Good catch, thanks! Done